run.js

```diff
@@ -185,49 +150,25 @@ function prepareMainSettings() {
   if (!process.env.GRIST_SESSION_SECRET) {
     process.env.GRIST_SESSION_SECRET = invent('GRIST_SESSION_SECRET');
   }
-
-  // When not using https either manually or via automation, the user
-  // presumably will tolerate cookies sent without https. See:
-  //   https://community.getgrist.com/t/solved-local-use-without-https/2852/11
-  if (!process.env.HTTPS && !process.env.INSECURE_COOKIE) {
-    // see https://github.com/thomseddon/traefik-forward-auth for
-    // documentation. This environment variable will be set when
-    // the traefik-forward-auth process is started (and others too,
-    // but won't have an impact on them).
-    process.env.INSECURE_COOKIE = 'true';
-  }
 }
 
 function prepareNetworkSettings() {
   const url = new URL(process.env.URL);
   process.env.APP_HOST = url.hostname || 'localhost';
-  // const extPort = parseInt(url.port || '9999', 10);
-  const extPort = url.port || '9999';
-  process.env.EXT_PORT = extPort;
-
-  // traefik-forward-auth will try to talk directly to dex, so it is
-  // important that URL works internally, withing the container. But
-  // if URL contains localhost, it really won't.  We can finess that
-  // by tying DEX_PORT to EXT_PORT in that case. As long as it isn't
-  // 80 or 443, since traefik is listening there...
-
   process.env.DEX_PORT = '9999';
-  if (process.env.APP_HOST === 'localhost' && extPort !== '80' && extPort !== '443') {
-    process.env.DEX_PORT = process.env.EXT_PORT;
-  }
 
   // Keep other ports out of the way of Dex port.
   const alt = String(process.env.DEX_PORT).charAt(0) === '1' ? '2' : '1';
   process.env.GRIST_PORT = `${alt}7100`;
-  process.env.TFA_PORT = `${alt}7101`;
-  process.env.WHOAMI_PORT = `${alt}7102`;
-
-  setBrittleEnv('DEFAULT_PROVIDER', 'oidc');
-  process.env.PROVIDERS_OIDC_CLIENT_ID = invent('PROVIDERS_OIDC_CLIENT_ID');
-  process.env.PROVIDERS_OIDC_CLIENT_SECRET = invent('PROVIDERS_OIDC_CLIENT_SECRET');
-  process.env.PROVIDERS_OIDC_ISSUER_URL = `${process.env.APP_HOME_URL}/dex`;
-  process.env.SECRET = invent('TFA_SECRET');
-  process.env.LOGOUT_REDIRECT = `${process.env.APP_HOME_URL}/signed-out`;
+  process.env.WHOAMI_PORT = `${alt}7101`;
+
+  // Setup OIDC
+  process.env.GRIST_OIDC_SP_HOST = process.env.APP_HOME_URL;
+  process.env.GRIST_OIDC_IDP_ISSUER = `${process.env.APP_HOME_URL}/dex`;
+  process.env.GRIST_OIDC_IDP_CLIENT_ID = invent('GRIST_OIDC_IDP_CLIENT_ID');
+  process.env.GRIST_OIDC_IDP_CLIENT_SECRET = invent('GRIST_OIDC_IDP_CLIENT_SECRET');
+  process.env.GRIST_OIDC_IDP_END_SESSION_ENDPOINT = `${process.env.APP_HOME_URL}/signed-out`;
```

Comment:
This URL should be an endpoint provided by the Identity provider to perform the log out. It looks like you redirect to the `signed-out` page of Grist?

I will setup a docker container, but if the issuer provides an `end_session_endpoint`, you should even omit that parameter, the logout would work out of the box.

Reply:
Without this variable:

```Error: The Identity provider does not propose end_session_endpoint. If that is expected, please set GRIST_OIDC_IDP_SKIP_END_SESSION_ENDPOINT=true or provide an alternative logout URL in GRIST_OIDC_IDP_END_SESSION_ENDPOINT
    at OIDCConfig.initOIDC (/grist/_build/app/server/lib/OIDCConfig.js:110:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.getMiddleware (/grist/_build/app/server/lib/OIDCConfig.js:229:13)
    at async FlexServer.addLoginMiddleware (/grist/_build/app/server/lib/FlexServer.js:1083:33)
    at async main (/grist/_build/app/server/mergedServerMain.js:86:5)
    at async main (/grist/_build/stubs/app/server/server.js:143:20)
error [grist-omnibus] grist failed: exited with 0
```

Reply:
Then you should probably go for `GRIST_OIDC_IDP_SKIP_END_SESSION_ENDPOINT=true`. But that also means that the user is still logged in in dex, and clicking again on `Sign in` would not ask for authentication.

By the way, I tried these steps:
```
$ make build
$ make run
```

I got this error with make run:
```
...
2024-08-26 16:59:29.198 - info: == Grist version is 1.1.14 (commit unknown)
2024-08-26 16:59:29.201 - debug: skipping incomplete language ar (set GRIST_OFFER_ALL_LANGUAGES if you want it)
2024-08-26 16:59:29.204 - debug: skipping incomplete language cs (set GRIST_OFFER_ALL_LANGUAGES if you want it)
2024-08-26 16:59:29.209 - debug: skipping incomplete language fa (set GRIST_OFFER_ALL_LANGUAGES if you want it)
2024-08-26 16:59:29.215 - debug: skipping incomplete language nl (set GRIST_OFFER_ALL_LANGUAGES if you want it)
2024-08-26 16:59:29.221 - debug: skipping incomplete language sk (set GRIST_OFFER_ALL_LANGUAGES if you want it)
2024-08-26 16:59:29.223 - debug: skipping incomplete language th (set GRIST_OFFER_ALL_LANGUAGES if you want it)
2024-08-26 16:59:29.234 - info: Loading empty config because /persist/config.json missing
Error: connect ECONNREFUSED 127.0.0.1:8484
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 8484
}
error [grist-omnibus] grist failed: exited with 0
make: *** [Makefile:13 : run] Erreur 1
```

FWIW, I get no result with this command: `sudo lsof -i tcp:8484`, so probably not due to port conflict.

Does it rings you a bell by chance?

Reply:
Hm, it does seem like a port conflict. This change makes `make run` work for me.

```diff
diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-PORT = 8484
+PORT = 9999
 TEAM = cool-beans
 
 # Possible bases: gristlabs/grist, or gristlabs/grist-ee
```

Reply:
Thank you @jordigh! This time the container starts, but when I try to go to `http://localhost:9999`, I get redirected to `https://localhost`.

If you want to continue the discussion elsewhere, we can, it does not relate to your PR obvisously :)

Reply:
FWIW, I found that issue, dex does not implement the `end_session_endpoint` indeed:
https://github.com/dexidp/dex/issues/1697

So you may either set `GRIST_OIDC_IDP_END_SESSION_ENDPOINT` the way you do or `GRIST_OIDC_IDP_SKIP_END_SESSION_ENDPOINT=true`.